source/world_builder/features/subducting_plate_models/temperature/mass_conserving.cc

```diff
@@ -260,26 +270,50 @@ namespace WorldBuilder
               const Objects::NaturalCoordinate trench_point_natural = Objects::NaturalCoordinate(trench_point,
                                                                       *(world->parameters.coordinate_system));
 
-              std::pair<double, double> ridge_parameters = Utilities::calculate_ridge_distance_and_spreading(mid_oceanic_ridges,
-                                                           plate_velocities_at_each_ridge_point,
-                                                           world->parameters.coordinate_system,
-                                                           trench_point_natural);
+              std::vector<double> ridge_parameters = Utilities::calculate_ridge_distance_and_spreading(mid_oceanic_ridges,
+                                                     ridge_spreading_velocities_at_each_ridge_point,
+                                                     world->parameters.coordinate_system,
+                                                     trench_point_natural,
+                                                     subducting_velocities,
+                                                     ridge_spreading_velocities.first);
```

Comment:
valgrind is tripping over this line, can you check if this is actually filled?

valgrind output for `gwb-dat tests/gwb-dat/cartesian_multiridge.wb tests/gwb-dat/cartesian_multiridge.dat --limit-debug-consistency-checks`
```
...
39: 1500e3 500e3 0 10e3 1538.93  0
39: ==4763== Invalid read of size 8
39: ==4763==    at 0x3C8AFC: WorldBuilder::Utilities::calculate_ridge_distance_and_spreading(std::vector<std::vector<WorldBuilder::Point<2>, std::allocator<WorldBuilder::Point<2> > >, std::allocator<std::vector<WorldBuilder::Point<2>, std::allocator<WorldBuilder::Point<2> > > > >, std::vector<std::vector<double, std::allocator<double> >, std::allocator<std::vector<double, std::allocator<double> > > >, std::unique_ptr<WorldBuilder::CoordinateSystems::Interface, std::default_delete<WorldBuilder::CoordinateSystems::Interface> > const&, WorldBuilder::Objects::NaturalCoordinate const&, std::vector<std::vector<double, std::allocator<double> >, std::allocator<std::vector<double, std::allocator<double> > > >, std::vector<double, std::allocator<double> >) (utilities.cc:1337)
39: ==4763==    by 0x37232B: WorldBuilder::Features::OceanicPlateModels::Temperature::PlateModel::get_temperature(WorldBuilder::Point<3> const&, WorldBuilder::Objects::NaturalCoordinate const&, double, double, double, double, double) const (plate_model.cc:168)
39: ==4763==    by 0x35DF6D: WorldBuilder::Features::OceanicPlate::properties(WorldBuilder::Point<3> const&, WorldBuilder::Objects::NaturalCoordinate const&, double, std::vector<std::array<unsigned int, 3ul>, std::allocator<std::array<unsigned int, 3ul> > > const&, double, std::vector<unsigned long, std::allocator<unsigned long> > const&, std::vector<double, std::allocator<double> >&) const (oceanic_plate.cc:176)
39: ==4763==    by 0x3CE692: WorldBuilder::World::properties(std::array<double, 3ul> const&, double, std::vector<std::array<unsigned int, 3ul>, std::allocator<std::array<unsigned int, 3ul> > > const&) const (world.cc:387)
39: ==4763==    by 0x31450E: main (main.cc:305)
39: ==4763==  Address 0x5f82118 is 0 bytes after a block of size 8 alloc'd
39: ==4763==    at 0x4849013: operator new(unsigned long) (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
39: ==4763==    by 0x43341D: __gnu_cxx::new_allocator<double>::allocate(unsigned long, void const*) (new_allocator.h:127)
39: ==4763==    by 0x4265D3: std::allocator_traits<std::allocator<double> >::allocate(std::allocator<double>&, unsigned long) (alloc_traits.h:464)
39: ==4763==    by 0x412877: std::_Vector_base<double, std::allocator<double> >::_M_allocate(unsigned long) (stl_vector.h:346)
39: ==4763==    by 0x3FFBD8: void std::vector<double, std::allocator<double> >::_M_range_initialize<double const*>(double const*, double const*, std::forward_iterator_tag) (stl_vector.h:1582)
39: ==4763==    by 0x3EBB6B: std::vector<double, std::allocator<double> >::vector(std::initializer_list<double>, std::allocator<double> const&) (stl_vector.h:629)
39: ==4763==    by 0x3721F9: WorldBuilder::Features::OceanicPlateModels::Temperature::PlateModel::get_temperature(WorldBuilder::Point<3> const&, WorldBuilder::Objects::NaturalCoordinate const&, double, double, double, double, double) const (plate_model.cc:168)
39: ==4763==    by 0x35DF6D: WorldBuilder::Features::OceanicPlate::properties(WorldBuilder::Point<3> const&, WorldBuilder::Objects::NaturalCoordinate const&, double, std::vector<std::array<unsigned int, 3ul>, std::allocator<std::array<unsigned int, 3ul> > > const&, double, std::vector<unsigned long, std::allocator<unsigned long> > const&, std::vector<double, std::allocator<double> >&) const (oceanic_plate.cc:176)
39: ==4763==    by 0x3CE692: WorldBuilder::World::properties(std::array<double, 3ul> const&, double, std::vector<std::array<unsigned int, 3ul>, std::allocator<std::array<unsigned int, 3ul> > > const&) const (world.cc:387)
39: ==4763==    by 0x31450E: main (main.cc:305)
39: ==4763== 
39: 1500e3 1000e3 0 10e3 500.274  0
...

```